src/layouts/BaseLayout.tsx

```diff
@@ -26,7 +26,7 @@ const GlobalStyle = createGlobalStyle`
 `;
 
 const BaseLayout = (): ReactElement => {
-  const user = useAtomValue(userAtomLoadable);
+  const user = useAtomValue(currentUserAtomLoadable);
 
   if (user.state === 'hasError') throw Error('Failed to fetch user data');
 
```

Comment:
```suggestion
  if (user.state === 'hasError') throw user.error;
```
どうやらこの行で伝搬された例外を新しい例外としてスローしているようです.  悪くないのですが, 懸念点があります.
たとえば, `src/stores/currentUserAtom.ts` に次のような例外をスローする処理を追加するとしましょう:

```diff
const fetchCurrentUser = async (): Promise<CurrentUser> => {
  await waitMs(2000);
+  throw new Error('これがエラーの原因だよ！');

  const user: CurrentUser = {
    id: '0000-0000-0000-0000',
    firstName: '智',
    // ...
```

しかし, L31 のままだとどれだけスタックトレースを遡っても `これがエラーの原因だよ！` にたどり着くことができないです.  
なので, 提案するように Loadable の `error` メンバーに Error が入っているのでそれを Re-throw するのはどうでしょう？

または `Error` オブジェクトには第 2 引数のオブジェクトに [`cause` プロパティー - mdn docs ↗](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) があるので次のように書いてエラーを Wrap しても良いかもしれません.  
そうすることでスタックトレースで遡っても, 両方見ることができます.

```ts
  if (user.state === 'hasError') throw Error('Failed to fetch user data', { cause: user.error });
```

(その場合, `tsconfig.json` の `compilerOptions.lib` にある `ES2020` を `ES2022` に変更してください.)

ref:
- [Error の cause オプションによってエラーの再 throw 時にスタックトレースが失われるのを防ぐ【ES2022】 - Qiita ↗](https://qiita.com/yamadashy/items/8f7455fb4d900853b60b)